pkg/jailer/jailer_test.go

```diff
@@ -0,0 +1,46 @@
+package jailer
+
+import (
+	"os"
+	"testing"
+
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_WhitelistEnvvars(t *testing.T) {
+	settings.WhitelistEnvironmentVars.Set("ENVVAR_FOO")
+	defer os.Unsetenv("ENVVAR_FOO")
+
+	type testCase struct {
+		input    []string
+		expected []string
+		envValue string
+	}
+
+	testCases := []testCase{
+		// Base case, nothing goes in, no env set, nothing comes out
+		{
+			input:    []string{},
+			expected: []string{},
+			envValue: "",
+		},
+		{
+			input:    []string{},
+			expected: []string{"ENVVAR_FOO=BAR"},
+			envValue: "BAR",
+		},
+		{
+			input:    []string{"BAZ=FOOBAZ"},
+			expected: []string{"BAZ=FOOBAZ", "ENVVAR_FOO=ALPHA"},
+			envValue: "ALPHA",
+		},
+	}
+
+	for _, tc := range testCases {
+		os.Setenv("ENVVAR_FOO", tc.envValue)
```

Comment:
Ridiculous nit but could the test cases be ordered:
{
envValue: "",
input:"",
expected:""
}

Reply:
Done